apps/sales/views.py

```diff
@@ -154,4 +157,168 @@ def create(self, request, *args, **kwargs):
+
+            # create data in 'order_shipments' model
+            provided_val = order_shipments_data.get('order_type_id')
+            get_id = OrderTypes.objects.all()
+            name_id_dictionary = {}
+            for obj in get_id:
+                name = obj.name
+                id = obj.order_type_id
+                name_id_dictionary[name] = id
+            order_shipments_data['order_id'] = sale_order_id
+            order_shipments_data['order_type_id'] = name_id_dictionary.get(provided_val)
+
+            serializer = OrderShipmentsSerializer(data=order_shipments_data)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+                data_4 = serializer.data   
+
+            if data_1 and data_2 and data_3 and data_4:  
+                custom_data = {
+                    "sale_order": data_1,
+                    "sale_order_items": data_2,
+                    "order_attachments":data_3,
+                    "order_shipments": data_4
+                }
+                return Response({
+                    'status': True,
+                    'message': 'Record created successfully',
+                    'data': custom_data
+                },status=status.HTTP_201_CREATED)
+            else:
+                return Response({
+                    'status': False,
+                    'message': 'Form validation failed',
+                    'errors': serializer.errors
+                }, status=status.HTTP_400_BAD_REQUEST)       
+
+    def retrieve(self, request, *args, **kwargs):
+        pk = kwargs.get('pk')  # Access the pk from kwargs
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+
+        # Query SaleOrderItems model using the pk
+        items_related_data = SaleOrderItems.objects.filter(sale_order_id=pk)  # 'sale_order_id' is the FK field
+        items_related_serializer = SaleOrderItemsSerializer(items_related_data, many=True)
+
+        # get sale_order_id value from SaleOrder Instance 
+        order_id = serializer.data.get('sale_order_id')
+
+        # Query OrderAttachments model using the order_id
+        attachments_related_data = OrderAttachments.objects.filter(order_id=str(pk))
+        attachments_related_serializer = OrderAttachmentsSerializer(attachments_related_data, many=True)
+
+        # Query OrderShipments model using the order_id
+        shipments_related_data = OrderShipments.objects.filter(order_id=str(order_id))
+        shipments_related_serializer = OrderShipmentsSerializer(shipments_related_data, many=True)
+
+        # Customizing the response data
+        custom_data = {
+            "sale_order": serializer.data,
+            "sale_order_items": items_related_serializer.data,
+            "order_attachments":attachments_related_serializer.data,
+            "order_shipments": shipments_related_serializer.data,
+        }
+        return Response(custom_data)
```

Comment:
The response data should match to out JSON pattern

Reply:
Updated Thulasi